File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def convert(n, k):
    tmp = n
    convert_num = ''
    while True:
        if tmp < k:
            convert_num = str(tmp) + convert_num
            break
        leaves = tmp % k
        tmp = tmp // k
        convert_num = str(leaves) + convert_num
    return convert_num

def is_prime(n):
    if n == 1: return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True

def solution(n, k):
    answer = 0    
    number = convert(n, k)
    tmp = number.split('0')

    for t in tmp:
        if t and is_prime(int(t)):
            answer += 1

    return answer

File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public static String convert(int n, int k) {
        int tmp = n;
        String convert_num = "";

        while(true) {
            if (tmp < k) {
                convert_num = tmp + convert_num;
                break;
            }
            int leaves = tmp % k;
            tmp = tmp / k;
            convert_num = leaves + convert_num;
        }
        return convert_num;
    }

    public static boolean is_prime(long n) {
        if (n == 1)
            return false;
//        System.out.println(Math.sqrt(n));
        for(long i = 2; i < Math.floor(Math.sqrt(n)) + 1; i++) {
            if (n % i == 0)
                return false;
        }
        return true;
    }

    public static int solution(int n, int k) {
        int answer = 0;
        String number = convert(n, k);
        String[] arr = number.split("0");
        for (String a: arr) {
            if (!a.isEmpty() && is_prime(Long.parseLong(a))) {
                answer += 1;
            }
        }

        return answer;
    }
}